plugin/trino-cassandra/src/main/java/io/trino/plugin/cassandra/CassandraClientModule.java

```diff
@@ -92,80 +89,70 @@ public static CassandraSession createCassandraSession(CassandraClientConfig conf
         requireNonNull(config, "config is null");
         requireNonNull(extraColumnMetadataCodec, "extraColumnMetadataCodec is null");
 
-        Cluster.Builder clusterBuilder = Cluster.builder();
+        CqlSessionBuilder cqlSessionBuilder = CqlSession.builder();
+        ProgrammaticDriverConfigLoaderBuilder driverConfigLoaderBuilder = DriverConfigLoader.programmaticBuilder();
+        // allow the retrieval of metadata for the system keyspaces
+        driverConfigLoaderBuilder.withStringList(DefaultDriverOption.METADATA_SCHEMA_REFRESHED_KEYSPACES, List.of());
+
         if (config.getProtocolVersion() != null) {
-            clusterBuilder.withProtocolVersion(config.getProtocolVersion());
+            driverConfigLoaderBuilder.withString(DefaultDriverOption.PROTOCOL_VERSION, config.getProtocolVersion().name());
         }
 
         List<String> contactPoints = requireNonNull(config.getContactPoints(), "contactPoints is null");
         checkArgument(!contactPoints.isEmpty(), "empty contactPoints");
-        clusterBuilder.withPort(config.getNativeProtocolPort());
-        clusterBuilder.withReconnectionPolicy(new ExponentialReconnectionPolicy(500, 10000));
-        clusterBuilder.withRetryPolicy(config.getRetryPolicy().getPolicy());
 
-        LoadBalancingPolicy loadPolicy = new RoundRobinPolicy();
+        driverConfigLoaderBuilder.withString(DefaultDriverOption.RECONNECTION_POLICY_CLASS, com.datastax.oss.driver.internal.core.connection.ExponentialReconnectionPolicy.class.getName());
+        driverConfigLoaderBuilder.withDuration(DefaultDriverOption.RECONNECTION_BASE_DELAY, Duration.ofMillis(500));
+        driverConfigLoaderBuilder.withDuration(DefaultDriverOption.RECONNECTION_MAX_DELAY, Duration.ofMillis(10_000));
+        driverConfigLoaderBuilder.withString(DefaultDriverOption.RETRY_POLICY_CLASS, config.getRetryPolicy().getPolicyClass().getName());
 
+        driverConfigLoaderBuilder.withString(DefaultDriverOption.LOAD_BALANCING_POLICY_CLASS, DefaultLoadBalancingPolicy.class.getName());
```

Comment:
The approach for setting the load balancing policy in the v4 version of the driver implies to rethink the appropriate settings for the cassandra plugin.

Have a look over https://github.com/datastax/java-driver/blob/4.x/core/src/main/resources/reference.conf for the available options for load balancing policy:

- `DefaultLoadBalancingPolicy`
- `BasicLoadBalancingPolicy`
- `DcInferringLoadBalancingPolicy`
- ...

NOTE that the current implementation hardcodes the usage of `DefaultLoadBalancingPolicy` load balancing policy.